src/containers/widget/index.js

```diff
@@ -45,12 +47,23 @@ class Widget extends Component {
   }
 
   componentDidMount() {
-    this.setDefaults();
+    return fetch(`${this.props.apiBaseUrl}/v1/configuration`,
+        { headers: {"Api-Token": this.props.apiToken} })
+      .then(response => {
+        if(response.ok) {
+          response.json().then(config => {
+            this.widgetConfig = config;
```

Comment:
Why are we storing it in a class variable here?

Reply:
to be able to set the initial state again (e.g. when the widget is closed) 
any idea how we else should do that?

Reply:
Whatever the code you are putting it in componentDidMount, that will be executed every time when that component **appears** on the browser.

So, there's no point in storing the response in this.widgetConfig unless you are adding a null check on componentDidMount before making an API call.

Reply:
can I execute something before it appears? 

I want the call to be made once and then it should be used similar to the props (set in the HTML).